lib/private/User/SyncService.php

```diff
@@ -248,13 +248,13 @@ private function syncHome(Account $a, UserInterface $backend) {
 			$existing = $a->getHome();
 			$backendHome = $backend->getHome($uid);
 			$class = \get_class($backend);
-			if ($existing !== '') {
+			if ($existing) {
```

Comment:
```suggestion
			if ($existing !== $backendHome) {
```
When I read the message, I feel like the test should be like what I suggest above.
Is there some reason why the test is just about `$existing` and does not involve `$backendHome` ?

Reply:
https://github.com/owncloud/core/blob/3f6b17300a4a24c3d286185d0d2c1cfdb6c58d46/lib/private/User/SyncService.php#L247

This control already exists in the upper level. It is the outer condition of this condition.